examples/download_sample_data.py

```diff
@@ -217,6 +233,40 @@ def _process_dataset(dataset, output_dir):
                 zipf.extractall()
             os.remove(output_path)
 
+
+def _unzip_7z(fname, delete_7z=True):
+    try:
+        arc = Archive7z(open(fname, 'rb'))
+    except:
+        print('FAILED ON 7Z', fname)
+        raise
+    fnames = arc.filenames
+    files = arc.files
+    data_dir = os.path.dirname(fname)
+    for fn, fi in zip(fnames, files):
+        gnd = path.join(data_dir, os.path.basename(fn))
+        if not os.path.exists(os.path.dirname(gnd)):
+            os.mkdir(os.path.dirname(gnd))
+        with open(gnd, 'w') as f:
+            f.write(fi.read().decode())
+        if delete_7z:
+            os.remove(fname)
+
+
+def download_puget_sound_lidar(here, files, url, **dataset):
+    urls = [url.replace('.html', '/' + _.strip())
+            for _ in files]
+    output_paths = [path.join(here, 'data', os.path.basename(fname))
+                    for fname in files]
+    title_fmt = 'Puget Sound Lidar {} of {}'
+    for idx, (url, output_path) in enumerate(zip(urls, output_paths)):
+        if os.path.exists(output_path.replace('7z', 'gnd')):
+            print('Skipping {0}'.format(title_fmt.format(idx + 1, len(urls))))
+            continue
+        _url_to_binary_write(url, output_path, title_fmt.format(idx + 1, len(urls)))
+        _unzip_7z(output_path, delete_7z=True)
+
```

Comment:
I went through a documented all the funcs in `download_sample_data.py` and tried to simplify more.  There are now two ways a dataset may be specified in datasets.yml:

* Same `files, url, title` keys pattern we already had in there.  This means download an archive from `url` if not all `files` are present already, then unpack the contents from `url` to create the `files`.
* `files, url, title, tag` keys pattern used by this lidar data example.  `tag` is used to separate special cases.  In the `tag="lidar"` case, a util is called to replace the url pattern to form 25 urls of 7z files, then download each and un7zip each.  The alternative to this special logic would have been to make 25 entries in `datasets.yml` and leave `download_sample_data.py` as it was, but I thought that would make `datasets.yml` less clear.